festim/hydrogen_transport_problem.py

```diff
@@ -126,22 +126,62 @@ def initialise(self):
     def defing_export_writers(self):
         """Defines the export writers of the model"""
         for export in self.exports:
-            # TODO implement when export.field is an int or str
-            # then find solution from index of species
+            # if name of species is given then replace with species object
+            for idx, field in enumerate(export.field):
+                if isinstance(field, str):
+                    export.field[idx] = F.find_species_from_name(field, self.species)
 
             if isinstance(export, (F.VTXExport, F.XDMFExport)):
                 export.define_writer(MPI.COMM_WORLD)
                 if isinstance(export, F.XDMFExport):
                     export.writer.write_mesh(self.mesh.mesh)
 
     def define_function_space(self):
```

Comment:
Now that this method has grown a bit, maybe it's time to describe the behaviour in docstrings